<launch>

  <!-- 障碍模型 -->
  <node pkg="gazebo_ros" type="spawn_model" name="spawn_box_1" output="screen"
        args="-sdf -file /media/fengyuanliu/extra_data/home_fengyuanliu/catkin_ws/src/vision_lyapunov_avoidance/models/box/box.sdf -sdf -model box_obstacle_1 -x 7 -y -3 -z 0"/>
  <node pkg="gazebo_ros" type="spawn_model" name="spawn_box_2" output="screen"
        args="-sdf -file /media/fengyuanliu/extra_data/home_fengyuanliu/catkin_ws/src/vision_lyapunov_avoidance/models/box/box.sdf -sdf -model box_obstacle_2 -x 9 -y 0 -z 0"/>
  <node pkg="gazebo_ros" type="spawn_model" name="spawn_box_3" output="screen"
        args="-sdf -file /media/fengyuanliu/extra_data/home_fengyuanliu/catkin_ws/src/vision_lyapunov_avoidance/models/box/box.sdf -sdf -model box_obstacle_3 -x 5 -y 0 -z 0"/>
      
  <!-- 避障节点 -->
  <node pkg="vision_lyapunov_avoidance" type="min_avoid_node.py" name="min_avoid" output="screen">
    <param name="depth_topic"    value="/iris/depth_camera/depth/image_raw"/>
    <param name="pred_enabled"   value="true"/>
    <param name="mpc_horizon"    value="1.2"/>
    <param name="mpc_num_angles" value="9"/>
    <param name="mpc_max_deg"    value="35.0"/>
    <param name="mpc_w_clear"    value="1.2"/>
    <param name="mpc_w_prog"     value="0.6"/>
    <param name="mpc_v_scale"    value="1.0"/>
    <param name="safe_distance"  value="1.8"/>
    <param name="avoid_speed"    value="0.55"/>
    <param name="forward_speed"  value="0.30"/>
    <param name="release_margin" value="0.4"/>    <!-- 防抖的释放阈值 -->
    <param name="alpha"          value="0.85"/>    <!-- 深度低通 -->
    <param name="rate"       value="30"/>     <!-- 发布频率Hz -->
    <param name="vmax_xy"        value="1.0"/>    <!-- 速度限幅 -->
    <param name="react_tau"      value="0.8"/>
    <param name="k_side"         value="0.7"/>
    <param name="k_forward"      value="0.5"/>
    <param name="use_7_sectors" value="true"/>
    <param name="ttc_thresh"    value="3.0"/>
    <param name="cbf_relax_eps" value="0.05"/>
    <param name="range_cap" value="6.0"/>
    <param name="asym_bias_deg" value="4.0"/>
    <param name="window" value="120"/>        
    <param name="k_side" value="0.6"/>
    <param name="k_forward" value="0.9"/>

  </node>
    
  <node pkg="vision_lyapunov_avoidance" type="path_accum.py" name="path_accum" output="screen"> 
    <param name="frame" value="world"/> 
  </node>
  
</launch>

